docs/DeveloperGuide.md

```diff
@@ -1,38 +1,208 @@
+The following resources and libraries were referenced or adapted in this project:
+* Java Coding Standard (Basic): [Official Documentation](https://se-education.org/guides/conventions/java/basic.html)
+* Code Quality (Guide): (https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html)
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Main Components of the architecture
+
+The `Inventra` class manages the overall application flow, which includes application launch and shutdown.
+The core processes are managed by the following components:
+* **UI**: The user interface for handling user interaction and the display of output.
+* **Command**: The commands that user inputs.
+* **CommandParser**: A parser to handle said commands, as well as interpreting flags and arguments.
+* **Inventory**: The mode we are operating in.
+* **Storage**: Reads and writes data to a CSV file to maintain data persistence.
+
+### "Add" Command Feature
+The "AddCommand" is responsible for adding fields and records to the inventory. 
+Concurrently, "add" command is implemented with behaviour to update CSV file to persist changes made to inventory.
+
+This command supports the following options:
+* '-h': Add custom fields to the inventory.
+* '-d': Add records to the inventory corresponding to defined fields.
+
+#### Architecture Overview
+The 'AddCommand' is part of the `command` package, and interacts with the following key components:
+* **Inventory**: Contains the current state of the inventory, including fields and records.
+* **Csv**: Handles reading from and writing to the CSV file for data storage persistence.
+* **Ui**: Handles user interactions component such as message to user during command execution.
+
+Command flow of 'AddCommand' can be visualized with the following architecture diagram:
+![Architecture Diagram](docs/diagrams/AddCommandArchitectureDiagram.png)
+
+* **Description**: The `AddCommand` class modifies the `Inventory` by adding new fields or records based on user's inputs.
+It interacts with `Csv` to update the CSV file accordingly to maintain data persistence. 
+The `Ui` components then displays feedback to the user.
+
+#### Component-Level Design
+"AddCommand" Class:
+* processes the input arguments and executes the logic based on flags (-h, -d) respectively.
+* fields and records are validated before updating inventory data into the CSV file.
+
+"AddCommand" Methods:
+* handleAddMultipleFields(): processes input fields, validate fields, and update the inventory accordingly.
+* handleAddRecord(): handles logic for adding records into the inventory. If successfully add records, data will be appended into the CSV file.
+* handleUpdateFields(): allow updating of existing fields and types in the inventory.
+  ![Class Diagram](docs/umldiagrams/AddCommandClassDiagram.png)
```

Comment:
For all diagrams, it would be better to show the diagram first before explaining the details within the diagram.